python/cudf/cudf/utils/string.py

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
```

Comment:
I like the name `utils/string.py` but it could be confused with utilities meant for use with string columns. I don't particularly have a better suggestion, just curious to see if other folks feel the same.

Reply:
I would just put this with the other utils and avoid the extra indirection here.

Reply:
I created `utils/string.py` to avoid cyclic dependencies since `utils/utils.py` imports buffer:
```python 
import cudf
from cudf.core import column
from cudf.core.buffer import as_device_buffer_like
```

I was thinking of adding it to `utils/docutils.py` but I think that would also be a bit confusing. 

What about creating `utils/config.py` for string and config utilities? E.g. the spilling manager also needs string parsing for functions such as [`_env_get_int()`](https://github.com/rapidsai/cudf/blob/d1317a64a6f22c836a887f3563f0e7add9884649/python/cudf/cudf/core/spill_manager.py#L153).

Reply:
I honestly prefer `utils/string.py` over `utils/config.py` as `format_bytes()` really doesn't do anything related to configuration.

One alternative here is to make the `Buffer` import local inside the function that needs it in `utils/utils.py`.